### ISSUE TITLE
LIMA_* variables are not set in boot provisioning script

### ISSUE THREAD

**Issue by jandubois** [lima-vm/lima#1685]
I tried:
```
- mode: boot
  script: |
    #!/bin/bash
    set -eux
    usermod -aG systemd-journal $LIMA_CIDATA_USER
```

This is the result:
```
/run/cloud-init/tmp/tmprbrvm5x4.sh: 4: LIMA_CIDATA_USER: parameter not set
2023-07-26 18:15:03,094 - util.py[WARNING]: Failed to run bootcmd module bootcmd
2023-07-26 18:15:03,095 - util.py[WARNING]: Running module bootcmd (<module 'cloudinit.config.cc_bootcmd' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_bootcmd.py'>) failed
```

I also had to remove the `-o pipefail` option from the `set -eux -o pipefail` line that I had copied to g

_Originally posted by @msgilligan in https://github.com/lima-vm/lima/discussions/691#discussioncomment-6554640_

**Comment by msgilligan** [comment]
Note that even if the `LIMA_CIDATA_USER` variable were set, the command will fail because that user has not been created at the time the `boot` mode script runs.

**Comment by msgilligan** [comment]
I created Issue #1686 which seems like a better approach to solving my specific problem.

**Comment by jandubois** [comment]
> I created Issue #1686 which seems like a better approach to solving my specific problem

I agree, but let's keep this issue open, as not having access to the `LIMA_CODATA_*` variables in the `boot` script is still a surprise.
